services/api/mw2pdf/src/classes/PdfFactory.ts

```diff
@@ -1,51 +1,34 @@
 import { fileURLToPath } from 'url'
 import path from 'path'
 import fs from 'fs'
-import { v4 as uuidv4 } from 'uuid'
 import PdfMerger from 'pdf-merger-js'
 import PdfPrinter from 'pdfmake'
 import {
   PDFDocument,
   rgb,
   StandardFonts,
 } from 'pdf-lib'
-import { Pdf } from './Pdf'
+import { Pdf, PdfConstructorOptions } from './Pdf'
 
 // The below two lines are from
 // https://stackoverflow.com/questions/32705219/nodejs-accessing-file-with-relative-path/32707530#32707530
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = path.dirname(__filename);
 
-export class PdfGenerator {
+export class PdfFactory {
 
-  directory: any;
-
-  constructor(directory = '') {
-    Object.assign(
-      this,
-      { directory },
-    )
-  }
-
-  generatePdfPath() {
-    return `${this.directory}${uuidv4()}.pdf`
-  }
-
-  generatePdfObject(options = {}) {
-    return new Pdf(
-      this.generatePdfPath(),
-      options,
-    )
+  static generatePdfObject(options: PdfConstructorOptions): Pdf {
+    return new Pdf(options);
   }
 
-  async generateMergedPdf(pdfs, outPdf = this.generatePdfObject()) {
-    const merger = new PdfMerger()
-    pdfs.forEach(pdf => merger.add(pdf.path))
-    await merger.save(outPdf.path)
-    return outPdf
+  static async generateMergedPdf(pdfs: Array<Pdf>, outPdf: Pdf): Promise<Pdf> {
+    const merger = new PdfMerger();
+    pdfs.forEach(pdf => merger.add(pdf.filename));
+    await merger.save(outPdf.filename);
+    return outPdf;
```

Comment:
semicolons in this method (some which created changed lines that aren't related to the PR)

Reply:
Took out all the semis in this file as well!